From 3da24c3ca35e733b3571d9f28bc5c98b1de72bdf Mon Sep 17 00:00:00 2001
From: Frank Barchard <fbarchard@google.com>
Date: Fri, 14 Oct 2022 11:56:23 -0700
Subject: [PATCH] Detile vld for gcc build fix

- add {} around loaded register

Bug: libyuv:944
Change-Id: I0d916e37beb50bda0838e4867742eb7afa57e1cc
Reviewed-on: https://chromium-review.googlesource.com/c/libyuv/libyuv/+/3957634
Reviewed-by: Justin Green <greenjustin@google.com>
Commit-Queue: Frank Barchard <fbarchard@chromium.org>
---
 README.chromium          |  2 +-
 include/libyuv/version.h |  2 +-
 source/row_neon.cc       | 12 ++++++------
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/source/row_neon.cc b/source/row_neon.cc
index c0db6f1d..b777a0e1 100644
--- a/source/row_neon.cc
+++ b/source/row_neon.cc
@@ -586,10 +586,10 @@ void DetileRow_NEON(const uint8_t* src,
                     int width) {
   asm volatile(
       "1:                                        \n"
-      "vld1.16     {q0}, [%0], %3                \n"  // load 16 bytes
+      "vld1.8      {q0}, [%0], %3                \n"  // load 16 bytes
       "subs        %2, %2, #16                   \n"  // 16 processed per loop
       "pld         [%0, #1792]                   \n"
-      "vst1.16     {q0}, [%1]!                   \n"  // store 16 bytes
+      "vst1.8      {q0}, [%1]!                   \n"  // store 16 bytes
       "bgt         1b                            \n"
       : "+r"(src),            // %0
         "+r"(dst),            // %1
@@ -632,9 +632,9 @@ void DetileToYUY2_NEON(const uint8_t* src_y,
                        int width) {
   asm volatile(
       "1:                                        \n"
-      "vld1.8      q0, [%0], %4                  \n"  // Load 16 Y
+      "vld1.8      {q0}, [%0], %4                \n"  // Load 16 Y
       "pld         [%0, #1792]                   \n"
-      "vld1.8      q1, [%1], %5                  \n"  // Load 8 UV
+      "vld1.8      {q1}, [%1], %5                \n"  // Load 8 UV
       "pld         [%1, #1792]                   \n"
       "subs        %3, %3, #16                   \n"
       "vst2.8      {q0, q1}, [%2]!               \n"
@@ -658,8 +658,8 @@ void DetileToYUY2_NEON(const uint8_t* src_y,
                        int width) {
   asm volatile(
       "1:                                        \n"
-      "vld1.8      q0, [%0], %4                  \n"  // Load 16 Y
-      "vld1.8      q1, [%1], %5                  \n"  // Load 8 UV
+      "vld1.8      {q0}, [%0], %4                \n"  // Load 16 Y
+      "vld1.8      {q1}, [%1], %5                \n"  // Load 8 UV
       "subs        %3, %3, #16                   \n"
       "pld         [%0, #1792]                   \n"
       "vzip.8      q0, q1                        \n"
-- 
2.25.1

